### ISSUE TITLE
Windows Terminal icon not showing up with "Invert Metro icon color" set to true

### ISSUE THREAD

**Issue by NebelNidas** [Open-Shell/Open-Shell-Menu#364]
**Describe the bug**
When turning on the "Invert Metro icon color" option, Open Shell leaves the colored icon background and instead removes the app icon itself, exactly the opposite of what it should do. I only experienced this with the newly released Windows Terminal until now, but if it happens there, it may occur with other apps as well.

**To Reproduce**
1. Install Windows Terminal from the MS Store: https://aka.ms/terminal
2. Turn on the "Invert Metro icon color" option in Open Shell
3. Search for the Terminal in Open Shell
4. See corrupted icon

**Expected behavior**
The (in my case) blue background should get removed, not the Terminal icon itself. 

**Screenshots**
"Invert Metro icon color" turned off (default Win10 behavior):
![image](https://user-images.githubusercontent.com/48808497/83498973-e7eb8400-a4bc-11ea-8dff-4b5804afcba2.png)
"Invert Metro icon color" turned on (issue):
![image](https://user-images.githubusercontent.com/48808497/83499020-f76acd00-a4bc-11ea-9ba8-3bccca5ec2e6.png)

**Version:**
 - Open-Shell: 4.4.142
 - OS: Windows 10 1909 and 2004, others probably too

**My thoughts:**
Maybe Open Shell tries to invert the colors, but because it's multi-colored and not black/white like the other icons, it fails? Thus not returning an inverted icon at all and leaving the background there because the method got interrupted? I may be completely wrong there, I unfortunately don't know any C++ to look for the problem myself, but that's what I think could be happening. 🤷‍♂️

**Comment by blackcrack** [comment]
Open Shell Corrupt no icons,
this makes MS by self.. and is not a Issue on Open Shell !

Because this reason beg i you, close this issue
because comes not from Open-Shell and prove you not knowing aboud Orderingsystems and User-Programms.. Open Shell is not responsible for system icons
So, please close this issue..
(by side of the "Comment Button !")

**Comment by NebelNidas** [comment]
> ... and prove you not knowing aboud Orderingsystems and User-Programms.. And the Fox is the badesd User-Picture what you have choose..

Wow, thanks for insulting me! I already said above that I'm not too familiar with C++, but I sure as hell know that this particular case is an issue with Open Shell, as it only happens when turning on the "Invert Metro icon color" option. So no, I'm not going to close this issue because a troll like you tells me to. And what does my profile picture have to do with this?!

**Comment by blackcrack** [comment]
>Wow, thanks for insulting me! 
wow you shoot hard ! alter, geht's noch ?! "Dude, you well ?!"
if you be so aggressive should be the post directly close from admin !
@Ibuprophen @ge0rdi @coddec 

> 
> 
> > ... and prove you not knowing aboud Orderingsystems and User-Programms.. And the Fox is the badesd User-Picture what you have choose..
> 
> Wow, thanks for insulting me! I already said above that I'm not too familiar with C++, but I sure as hell know that this particular case is an issue with Open Shell, as it only happens when turning on the "Invert Metro icon color" option. So no, I'm not going to close this issue because a troll like you tells me to. And what does my profile picture have to do with this?!

**Comment by ge0rdi** [comment]
Guys, please, calm down. There is no need for rough words here.
The report seems pretty valid to me.

@blackcrack 
Please, will be great to not tell everybody to close their issues without even investigating them.
If you want to help, then please try to investigate what's going on, ask user for additional info/clarification.
If you are unable to replicate the issue, that's fine. Not everything is easily replicable on other machines.

**Comment by blackcrack** [comment]
@ge0rdi, since the icons are read in, I think that it will certainly not be an Open Shell problem.
Orr ? :)

**Comment by ge0rdi** [comment]
@blackcrack 
`Invert Metro icon color` is setting of `Open-Shell`. It determines how `Open-Shell` displays icons.
Thus if icon looks weird with the option set and it is Ok without it, it looks like some problem in `Open-Shell` to me.

I haven't checked the code, but so far I don't see any reason why it shouldn't be related to `Open-Shell`.

**Comment by blackcrack** [comment]
okey... step back... therefore a Sorry for @NebelNidas ..

**Comment by NebelNidas** [comment]
Yeah, sorry for the rough words on my part, too. I just didn't like how you immediately started insulting me on a personal level for just a simple issue report while not even investigating it. But this isn't a discussion forum, people just try to get things done here on a neutral level, so let's just both try to be a little more formal next time around. :)

**Comment by NebelNidas** [comment]
I've tracked down the code location: [Open-Shell-Menu/blob/master/Src/StartMenu/StartMenuDLL/ItemManager.cpp#L207](https://github.com/Open-Shell/Open-Shell-Menu/blob/master/Src/StartMenu/StartMenuDLL/ItemManager.cpp#L207) However, as I'm not yet familiar with Open Shell's structure and C++code in general, I'm not really able to judge what is causing the issue here... 😕

**Comment by CTVCAM8** [comment]
Hello
I don't use that many apps.
I looked earlier in the day at all the app's icons using the older Classic Shell start menu.
All looked correct with a normal and inverted view (colored opaque symbols on transparent background). 
The Terminal one is just like yours,(wrong opaque rectangular color).


When the Terminal app icon is pinned to the taskbar, it shows correctly in the unpin menu view when dark, light, and accent color-tinted modes. 
So maybe that got fixed already. 

I also looked briefly at some other GitHub Terminal pages about profile icons and icon replacement.
Sorry, I never saved the pages. 
I read that someone mentioned that all of the png icons won't show its transparency correctly and show it as a white background.
Maybe there is a problem and then again could be that the person is seeing the rgb channels of icon or background in a photo image program.
Don't know. 
If it is related to the actual Terminal app icon misbehaving, 
Open/Classic shell seems to be doing exactly what it was designed to do. Anything that started as opaque will show as an opaque accent color.

**Comment by ge0rdi** [comment]
@NebelNidas @CTVCAM8 
I have created fix for this.
See #370, there is also some explanation (hopefully understandable).

Please, can you try [this build](https://ci.appveyor.com/project/passionate-coder/open-shell-menu/builds/33356383/artifacts) and report back if it works for you?

**Comment by NebelNidas** [comment]
Yes, it works! Thanks for the quick fix! 👍
![image](https://user-images.githubusercontent.com/48808497/83952698-bbc46000-a83a-11ea-8e86-a5defd608ead.png)

**Comment by CTVCAM8** [comment]
Same here works correctly. 
Than
![4 4 144 result](https://user-images.githubusercontent.com/44127216/83952863-e093e580-a7f0-11ea-9437-db9297e8ded7.png)
k you

**Comment by CTVCAM8** [comment]
Hi @NebelNidas. 
This is a different subject that excites me very much related to your first post.
In your screenshots of what looks like the 7 style internal programs tree menu area,
how on earth were you able to get a skinned scroll bar working again?
The programs scroll bar hasn't worked on Win 10 since Creators Update years ago.
I'm on Windows 10 1909
I tried Classic 4.3.1 and Open-Shell 4.4.142 versions. Skinned scroll bar only works on Win7 and Win8.1

**Comment by NebelNidas** [comment]
@CTVCAM8 Unfortunately only by installing a seperate Windows Theme (in my case the [Areo 7 theme for Windows 10](https://www.deviantart.com/sagorpirbd/art/Aero-7-Themes-for-Win10-Final-523979941https://www.deviantart.com/sagorpirbd/art/Aero-7-Themes-for-Win10-Final-523979941), which sadly isn't maintained anymore, however still mostly works). You need [UltraUXThemePatcher](https://www.syssel.net/hoefs/software_uxtheme.phprl) in order to get these custom styles to work, which might brick your Windows on some of the bigger Win10 updates if you're unlucky. But I think you as a theming expert know that already. :-)

**Comment by CTVCAM8** [comment]
I see. Oh, I've never done theming, just played around with skinning the start menu. 
I made the decision in Win 10 awhile ago to stop using themes as they kept breaking. Still have UltraUXThemePatcher installed.

I did put in a request way back to get scrollbar skinned again, or at least have an option to use dark mode scrollbar included, 
but looks like it fell on deaf ears as the limited amount of coders here have more pressing issues to deal with. 
Nice talking with you.
